File: src/CalendarPrinter/Logic/TagsToIconConverter.cs
using CalendarPrinter.Model;
using System;
using System.Collections.Generic;

namespace CalendarPrinter.Logic
{
    internal class TagsToIconConverter
    {
        private readonly List<TagMapping> _tagMapping;

        public TagsToIconConverter(List<TagMapping> tagMapping)
        {
            _tagMapping = tagMapping;
        }

        internal TagMapping GetIcon(List<string> tags)
        {
            if (tags.Count == 0)
                return null;

            foreach (var mapping in _tagMapping)
            {
                if (ContainsTags(mapping.Tags, tags))
                    return mapping;
            }

            return null;
        }

        private bool ContainsTags(List<string> needles, List<string> haystack)
        {
            foreach (var needle in needles)
            {
                if(!haystack.Contains(needle))
                    return false;
            }
            return true;
        }
    }
}

File: src/CalendarPrinter/TagsToIconConverter.py
class TagsToIconConverter:
    def __init__(self, mapping):
        self._mapping = mapping

    def GetIcon(self, tags):
        if len(tags) == 0:
            return None
        
        for mapping in self._mapping:
            if TagsToIconConverter.ContainsTags(mapping.tags, tags):
                return mapping
                
        return None
        
    def ContainsTags(needles, haystack):
        for needle in needles:
            if not needle in haystack:
                return False
        
        return True
